compiler/rustc_resolve/src/late.rs

```diff
@@ -1151,23 +1180,59 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                     .span_label(lifetime.ident.span, note)
                     .emit();
 
+                    self.r.lifetimes_res_map.insert(lifetime.id, LifetimeRes::Error);
+                    return;
+                }
+                LifetimeRibKind::AnonymousPassThrough(node_id) => {
+                    self.r
+                        .lifetimes_res_map
+                        .insert(lifetime.id, LifetimeRes::Anonymous { binder: node_id, elided });
                     return;
                 }
-                LifetimeRibKind::AnonymousCreateParameter
-                | LifetimeRibKind::AnonymousPassThrough
-                | LifetimeRibKind::Item => return,
+                LifetimeRibKind::Item => break,
                 _ => {}
             }
         }
+        self.r
+            .lifetimes_res_map
+            .insert(lifetime.id, LifetimeRes::Anonymous { binder: DUMMY_NODE_ID, elided });
     }
 
     #[tracing::instrument(level = "debug", skip(self))]
     fn resolve_elided_lifetime(&mut self, anchor_id: NodeId, span: Span) {
         let id = self.r.next_node_id();
+        self.r.lifetimes_res_map.insert(
+            anchor_id,
+            LifetimeRes::ElidedAnchor { start: id, end: NodeId::from_u32(id.as_u32() + 1) },
+        );
+
         let lt = Lifetime { id, ident: Ident::new(kw::UnderscoreLifetime, span) };
         self.resolve_anonymous_lifetime(&lt, true);
     }
 
+    #[tracing::instrument(level = "debug", skip(self))]
+    fn create_anonymous_lifetime(&mut self, id: NodeId, ident: Ident, item_node_id: NodeId) {
```

Comment:
```suggestion
    fn create_fresh_lifetime(&mut self, id: NodeId, ident: Ident, item_node_id: NodeId) {
```
There's a naming mismatch here, the function says "anonymous", but actually creates `LifetimeRes::Fresh` and not `LifetimeRes::Anonymous`.